lib/RewriteInsertsPass.cpp

```diff
@@ -137,6 +149,56 @@ class RewriteInsertsPass : public ModulePass {
      return nullptr;
    }
 
+   // If this is the tail of a chain of InsertElementInst instructions
+   // that covers the entire vector, then return a small vector
+   // containing the insertion instructions, in member order.
+   // Otherwise returns nullptr.  Only handle insertions into vectors.
+   InsertionVector *CompleteInsertionChain(InsertElementInst *ie) {
+     // Don't handle i8 vectors. Only <4 x i8> is supported and it is
+     // translated as i32.  Only handle single-index insertions.
+     if (auto vec_ty = dyn_cast<VectorType>(ie->getType())) {
```

Comment:
Since <4 x i8> is codegen'd as i32 in SPIRVProducerPass, I've disabled the transformation for vectors of i8 so that invalid SPIR-V is not produced.